# Data Science Project


In [11]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

In [12]:
df = pd.read_csv('players_22.csv', low_memory = False)
print(f"Dimensions : {df.shape[0]} joueurs et {df.shape[1]} variables")

Dimensions : 19239 joueurs et 110 variables


In [13]:
df.head()

,sofifa_id,player_url,short_name,long_name,player_positions,overall,potential,value_eur,wage_eur,age,...,lcb,cb,rcb,rb,gk,player_face_url,club_logo_url,club_flag_url,nation_logo_url,nation_flag_url
0,158023,https://sofifa.com/player/158023/lionel-messi/...,L. Messi,Lionel Andrés Messi Cuccittini,"RW, ST, CF",93,93,78000000.0,320000.0,34,...,50+3,50+3,50+3,61+3,19+3,https://cdn.sofifa.net/players/158/023/22_120.png,https://cdn.sofifa.net/teams/73/60.png,https://cdn.sofifa.net/flags/fr.png,https://cdn.sofifa.net/teams/1369/60.png,https://cdn.sofifa.net/flags/ar.png
1,188545,https://sofifa.com/player/188545/robert-lewand...,R. Lewandowski,Robert Lewandowski,ST,92,92,119500000.0,270000.0,32,...,60+3,60+3,60+3,61+3,19+3,https://cdn.sofifa.net/players/188/545/22_120.png,https://cdn.sofifa.net/teams/21/60.png,https://cdn.sofifa.net/flags/de.png,https://cdn.sofifa.net/teams/1353/60.png,https://cdn.sofifa.net/flags/pl.png
2,20801,https://sofifa.com/player/20801/c-ronaldo-dos-...,Cristiano Ronaldo,Cristiano Ronaldo dos Santos Aveiro,"ST, LW",91,91,45000000.0,270000.0,36,...,53+3,53+3,53+3,60+3,20+3,https://cdn.sofifa.net/players/020/801/22_120.png,https://cdn.sofifa.net/teams/11/60.png,https://cdn.sofifa.net/flags/gb-eng.png,https://cdn.sofifa.net/teams/1354/60.png,https://cdn.sofifa.net/flags/pt.png
3,190871,https://sofifa.com/player/190871/neymar-da-sil...,Neymar Jr,Neymar da Silva Santos Júnior,"LW, CAM",91,91,129000000.0,270000.0,29,...,50+3,50+3,50+3,62+3,20+3,https://cdn.sofifa.net/players/190/871/22_120.png,https://cdn.sofifa.net/teams/73/60.png,https://cdn.sofifa.net/flags/fr.png,NaN,https://cdn.sofifa.net/flags/br.png
4,192985,https://sofifa.com/player/192985/kevin-de-bruy...,K. De Bruyne,Kevin De Bruyne,"CM, CAM",91,91,125500000.0,350000.0,30,...,69+3,69+3,69+3,75+3,21+3,https://cdn.sofifa.net/players/192/985/22_120.png,https://cdn.sofifa.net/teams/10/60.png,https://cdn.sofifa.net/flags/gb-eng.png,https://cdn.sofifa.net/teams/1325/60.png,https://cdn.sofifa.net/flags/be.png


In [14]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 19239 entries, 0 to 19238
Columns: 110 entries, sofifa_id to nation_flag_url
dtypes: float64(16), int64(44), str(50)
memory usage: 16.1 MB


In [15]:
df.describe()

,sofifa_id,overall,potential,value_eur,wage_eur,age,height_cm,weight_kg,club_team_id,league_level,...,mentality_composure,defending_marking_awareness,defending_standing_tackle,defending_sliding_tackle,goalkeeping_diving,goalkeeping_handling,goalkeeping_kicking,goalkeeping_positioning,goalkeeping_reflexes,goalkeeping_speed
count,19239.000000,19239.000000,19239.000000,1.916500e+04,19178.000000,19239.000000,19239.000000,19239.000000,19178.000000,19178.000000,...,19239.000000,19239.000000,19239.000000,19239.000000,19239.000000,19239.000000,19239.000000,19239.000000,19239.000000,2132.000000
mean,231468.086959,65.772182,71.079370,2.850452e+06,9017.989363,25.210822,181.299704,74.943032,50580.498123,1.354364,...,57.929830,46.601746,48.045584,45.906700,16.406102,16.192474,16.055356,16.229274,16.491814,36.439962
std,27039.717497,6.880232,6.086213,7.613700e+06,19470.176724,4.748235,6.863179,7.069434,54401.868535,0.747865,...,12.159326,20.200807,21.232718,20.755683,17.574028,16.839528,16.564554,17.059779,17.884833,10.751563
min,41.000000,47.000000,49.000000,9.000000e+03,500.000000,16.000000,155.000000,49.000000,1.000000,1.000000,...,12.000000,4.000000,5.000000,5.000000,2.000000,2.000000,2.000000,2.000000,2.000000,15.000000
25%,214413.500000,61.000000,67.000000,4.750000e+05,1000.000000,21.000000,176.000000,70.000000,479.000000,1.000000,...,50.000000,29.000000,28.000000,25.000000,8.000000,8.000000,8.000000,8.000000,8.000000,27.000000
50%,236543.000000,66.000000,71.000000,9.750000e+05,3000.000000,25.000000,181.000000,75.000000,1938.000000,1.000000,...,59.000000,52.000000,56.000000,53.000000,11.000000,11.000000,11.000000,11.000000,11.000000,36.000000
75%,253532.500000,70.000000,75.000000,2.000000e+06,8000.000000,29.000000,186.000000,80.000000,111139.000000,1.000000,...,66.000000,63.000000,65.000000,63.000000,14.000000,14.000000,14.000000,14.000000,14.000000,45.000000
max,264640.000000,93.000000,95.000000,1.940000e+08,350000.000000,54.000000,206.000000,110.000000,115820.000000,5.000000,...,96.000000,93.000000,93.000000,92.000000,91.000000,92.000000,93.000000,92.000000,90.000000,65.000000


In [16]:
print(df.columns.tolist())

['sofifa_id', 'player_url', 'short_name', 'long_name', 'player_positions', 'overall', 'potential', 'value_eur', 'wage_eur', 'age', 'dob', 'height_cm', 'weight_kg', 'club_team_id', 'club_name', 'league_name', 'league_level', 'club_position', 'club_jersey_number', 'club_loaned_from', 'club_joined', 'club_contract_valid_until', 'nationality_id', 'nationality_name', 'nation_team_id', 'nation_position', 'nation_jersey_number', 'preferred_foot', 'weak_foot', 'skill_moves', 'international_reputation', 'work_rate', 'body_type', 'real_face', 'release_clause_eur', 'player_tags', 'player_traits', 'pace', 'shooting', 'passing', 'dribbling', 'defending', 'physic', 'attacking_crossing', 'attacking_finishing', 'attacking_heading_accuracy', 'attacking_short_passing', 'attacking_volleys', 'skill_dribbling', 'skill_curve', 'skill_fk_accuracy', 'skill_long_passing', 'skill_ball_control', 'movement_acceleration', 'movement_sprint_speed', 'movement_agility', 'movement_reactions', 'movement_balance', 'power

## Nettoyage du dataset

Suppression des colonnes inutiles. Les colonnes qui seront supprimées continnent des informations qui ne seront pas utilisées pour la modélisation, tel que:
 - les liens et images (`player_url`, `player_face_url`, ..)
 - les IDs (`sofifa_id`)
 - les informations spécifiques / administratives(`real_face`, `body_type`, ..)

In [17]:
cols_a_supprimer = [
    'player_url', 'player_face_url',
    'club_logo_url', 'club_flag_url', 'nation_logo_url', 'nation_flag_url',
    'sofifa_id',
    'real_face', 'body_type',
    'long_name', 'dob', 'club_team_id',
    'club_jersey_number', 'nationality_id', 'nation_team_id',
    'nation_jersey_number', 'nation_position', 'player_tags',
    'player_traits',
    'ls', 'st', 'rs', 'lw', 'lf', 'cf', 'rf', 'rw',
    'lam', 'cam', 'ram', 'lm', 'lcm', 'cm', 'rcm', 'rm',
    'lwb', 'ldm', 'cdm', 'rdm', 'rwb', 'lb', 'lcb',
    'cb', 'rcb', 'rb', 'gk'
]

df = df.drop(columns=cols_a_supprimer)

print(f"Colonnes restantes : {df.shape[1]}")
print(f"Joueurs : {df.shape[0]}")
print("\nListe finale des colonnes :")
print(list(df.columns))

Colonnes restantes : 64
Joueurs : 19239

Liste finale des colonnes :
['short_name', 'player_positions', 'overall', 'potential', 'value_eur', 'wage_eur', 'age', 'height_cm', 'weight_kg', 'club_name', 'league_name', 'league_level', 'club_position', 'club_loaned_from', 'club_joined', 'club_contract_valid_until', 'nationality_name', 'preferred_foot', 'weak_foot', 'skill_moves', 'international_reputation', 'work_rate', 'release_clause_eur', 'pace', 'shooting', 'passing', 'dribbling', 'defending', 'physic', 'attacking_crossing', 'attacking_finishing', 'attacking_heading_accuracy', 'attacking_short_passing', 'attacking_volleys', 'skill_dribbling', 'skill_curve', 'skill_fk_accuracy', 'skill_long_passing', 'skill_ball_control', 'movement_acceleration', 'movement_sprint_speed', 'movement_agility', 'movement_reactions', 'movement_balance', 'power_shot_power', 'power_jumping', 'power_stamina', 'power_strength', 'power_long_shots', 'mentality_aggression', 'mentality_interceptions', 'mentality_posit

## Valeurs manquantes

In [18]:
# Analyse des valeurs manquantes
valeurs_manquantes = df.isnull().sum()
valeurs_manquantes_pct = (df.isnull().sum() / len(df) * 100).round(2)

# tableau récapitulatif (décroissant)
missing_df = pd.DataFrame({
    'Valeurs manquantes': valeurs_manquantes,
    'Pourcentage (%)': valeurs_manquantes_pct
}).sort_values('Pourcentage (%)', ascending=False)

# colonnes qui ont au moins 1 valeur manquante
missing_df = missing_df[missing_df['Valeurs manquantes'] > 0]

print(f"Colonnes avec valeurs manquantes : {len(missing_df)}")
print()
print(missing_df)

Colonnes avec valeurs manquantes : 17

                           Valeurs manquantes  Pourcentage (%)
club_loaned_from                        18137            94.27
goalkeeping_speed                       17107            88.92
shooting                                 2132            11.08
passing                                  2132            11.08
physic                                   2132            11.08
pace                                     2132            11.08
dribbling                                2132            11.08
defending                                2132            11.08
release_clause_eur                       1176             6.11
club_joined                              1163             6.05
value_eur                                  74             0.38
club_contract_valid_until                  61             0.32
league_name                                61             0.32
club_name                                  61             0.32
league_level    

À supprimer — trop de valeurs manquantes
club_loaned_from → 94.27% manquant. Normal : seuls les joueurs en prêt ont cette info. Inutilisable, on supprime.
goalkeeping_speed → 88.92% manquant. Cette stat n'existe que pour les gardiens. On supprime.

Cas particulier — les 11.08% (pace, shooting, passing, dribbling, defending, physic)
Ces 6 colonnes ont exactement le même nombre de valeurs manquantes — 2132 joueurs. Ce n'est pas un hasard : ce sont les gardiens de but ! FIFA ne leur attribue pas ces stats générales. On a deux options :

Les supprimer → on perd de l'info sur les joueurs de champ
Les remplir par 0 pour les gardiens → logique métier : un gardien n'a pas de stat "shooting"

👉 On choisit de remplir par 0, c'est justifiable et on garde l'info.

🔧 À traiter intelligemment — les petits pourcentages
release_clause_eur (6.11%) et club_joined (6.05%) → on remplace par la médiane pour la clause et par une valeur neutre pour la date.
value_eur (0.38%) → c'est notre variable cible ! On ne peut pas inventer une valeur. On supprime ces 74 lignes.
Les 61 joueurs manquants sur club_name, league_name, league_level, club_position, club_contract_valid_until, wage_eur → ce sont probablement des joueurs sans club. On les supprime car ils représentent moins de 0.32% des données.

In [ ]:
 jf